### PR TITLE
docs: add basic doc entry for how nightly releases work and how they run

### PR DESCRIPTION
**Pull request by mcansh**
Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

File: docs/pages/contributing.md [hunk 1]
```diff
@@ -143,10 +143,16 @@ This repo maintains separate branches for different purposes. They will look som
 
 There may be other branches for various features and experimentation, but all of the magic happens from these branches.
 
+## How the heck do nightly releases work?
+
+Nightly releases will run the action files from the `main` branch as scheduled workflows will always use the latest commit to the default branch, signified by [this comment on the nightly action file][nightly-action-comment] and the explicit branch appended to the reuasable workflows in the [postrelease action][postrelease-action], however they clone the `dev` branch during their set up as that's where we want out nightly releases to be cut from. From there, we check if the git sha is the same and only cut a new nightly if something has changed.
```
Comment (MichaelDeBoey) [inline docs/pages/contributing.md:148]:
```suggestion
Nightly releases will run the action files from the `main` branch as scheduled workflows will always use the latest commit to the default branch, signified by [this comment on the nightly action file][nightly-action-comment] and the explicit branch appended to the reuasable workflows in the [postrelease action][postrelease-action], however they clone the `dev` branch during their set up as that's where we want our nightly releases to be cut from. From there, we check if the git sha is the same and only cut a new nightly if something has changed.
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Review by kentcdodds — APPROVED** [review]
This is great and highlights how awesome the work you've done is @mcansh. Amazing work 👍

**Comment by kentcdodds** [comment]
Is there a reason this is still a draft PR?

**Comment by mcansh** [comment]
> 

no reason other than i forgot to move when i updated linear 😆

**Pull request merged**